internal/controllers/topology/cluster/patches/variables/variables_test.go

```diff
@@ -450,7 +540,73 @@ func TestMachineDeployment(t *testing.T) {
 			},
 		},
 		{
-			name: "Should calculate MachineDeployment variables (without overrides)",
+			name:      "Should calculate MachineDeployment variables for a given patch name",
+			patchName: "patch1",
+			validDefinitions: map[string]bool{
+				"location": true,
+				"cpu":      true,
+			},
+			mdTopology: &clusterv1.MachineDeploymentTopology{
+				Replicas: pointer.Int32(3),
+				Name:     "md-topology",
+				Class:    "md-class",
+				Variables: &clusterv1.MachineDeploymentVariables{
+					Overrides: []clusterv1.ClusterVariable{
+						{
+							Name:           "location",
+							Value:          toJSON("\"us-central\""),
+							DefinitionFrom: "patch1",
+						},
+						{
+							Name:           "location",
+							Value:          toJSON("\"us-central\""),
```

Comment:
nit: maybe also add the same comment here as in the global test case (and also let's use a different value to be able to verify which one is picked)